public/app/plugins/panel/geomap/components/ComplexDataHoverView.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react';
+
+import { GeomapLayerHover } from '../event';
+import { DataHoverTabs } from './DataHoverTabs';
+import { DataHoverRows } from './DataHoverRows';
+import { CloseButton } from 'app/core/components/CloseButton/CloseButton';
+
+export interface Props {
+  layers?: GeomapLayerHover[];
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const ComplexDataHoverView = ({ layers, onClose, isOpen }: Props) => {
+  const [activeTabIndex, setActiveTabIndex] = useState<number>(0);
+
```

Comment:
```suggestion
if (!layers) {
  return null
}
```
we can avoid some nesting below